cli/src/commit_templater.rs

```diff
@@ -1671,3 +1723,76 @@ fn builtin_tree_diff_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, T
     // TODO: add files() or map() to support custom summary-like formatting?
     map
 }
+
+#[derive(Debug)]
+pub struct CryptographicSignature {
+    commit: Commit,
+}
+
+impl CryptographicSignature {
+    fn new(commit: Commit) -> Option<Self> {
+        commit.is_signed().then_some(Self { commit })
+    }
+
+    fn verify(&self) -> SignResult<Verification> {
+        self.commit
+            .verification()
+            .transpose()
+            .expect("must have signature")
+    }
+
+    fn status(&self) -> SignResult<SigStatus> {
+        self.verify().map(|verification| verification.status)
+    }
+
+    /// Defaults to empty string if key is not present.
+    fn key(&self) -> Result<String, TemplatePropertyError> {
+        self.verify()
+            .map_err(|e| TemplatePropertyError(e.into()))
+            .map(|verification| verification.key.unwrap_or_default())
+    }
```

Comment:
nit: for consistency, `key()` and `display()` can return `SignResult<String>`. The error can be mapped by caller as `Ok(sig.key()?)`.